09_maps/main.go

```diff
@@ -111,18 +122,129 @@ func main() {
+	// Close main function, to be able to declare another function
+}
 
-	// TODO: Create a function that takes an array of maps as an argument and
-	// returns the map with the highest BMI.
+// create empty map
+// create function
+// create counter
+// create loop
+// create break condition if counter is > than the key-value pairs the map has
+// operate on the input value of the keys and do math on it
+// add the result to a new map
+// add a new key value element and fill it depending on the value of the input height key
+// print new map
+
+var newMap map[string]int = map[string]int{}
+
+func createNewMap(mapInput map[string]int) {
+	var counter int = 0
+
+	for {
+		if counter > len(mapInput) {
+			break
+		}
+		newMap["age"] = mapInput["age"] + 2
+		newMap["height"] = mapInput["height"] * 2
+		newMap["weight"] = mapInput["weight"] * 2
+		if mapInput["height"] > 200 {
+			newMap["tall"] = 1
+		}
+		newMap["tall"] = 0
+		counter++
+	}
+	fmt.Println("createNewMap:", newMap)
+}
 
-	// TODO: Create a function that takes an array of maps as an argument and
-	// add as a new key-value pair to each map the key "bmi" with the value
-	// being the BMI of the corresponding person.
-	// Hint: Use a loop.
-	// Hint: Create a new map for each person.
-	// Hint: Create a new array of maps.
+// TODO: Create a function that takes a map as an argument and prints the
+// BMI of the person. The formula for BMI is:
+// BMI = weight / (height * height)
+// If the map does not have the key "height", print "Height not found".
+// If the map does not have the key "weight", print "Weight not found".
+// Otherwise, print "<name>'s BMI is <bmi>".
+// Hint: Use this function in later exercises.
+
+func calcBMI(mapInput map[string]int) {
+	var value int
+	var exists bool
+	var bmi float32
+	var nonFloatBmi float64
+	var heightAdjust float64
+
+	var heightConvert int = mapInput["height"]
+
+	var heighConverted float64 = float64(heightConvert)
+
+	value, exists = mapInput["height"]
+	if !exists || value == 0 {
+		fmt.Println("Height not found")
+	}
+	value, exists = mapInput["weight"]
+	if !exists || value == 0 {
+		fmt.Println("Weight not found")
+	}
+	// debug
+	fmt.Println("weight debug", mapInput["weight"])
+	fmt.Println("height debug", mapInput["height"])
+	// bmi = mapInput["weight"] / (mapInput["height"]/100 * mapInput["height"]/100)
+	// bmi = mapInput["weight"] / (heighAdjust * heighAdjust)
+	// heightConvert = float64((mapInput["height"])
+	heightAdjust = heighConverted / 100
+	// bmi = (mapInput["weight"]) / heighAdjust * heighAdjust
+	// bmi = 70 / ((180 / 100) * (180 / 100))
+	// bmi = (70 / (1.80 * 1.80))
+	nonFloatBmi = (70 / (180 * 180))
+	// debug
+	fmt.Println("nonfloatBMI is", nonFloatBmi)
+	nonFloatBmi = nonFloatBmi * 10000
+	fmt.Println("nonfloatBMI is", nonFloatBmi)
+
+	// debug
+	// fmt.Println(bmi)
+	fmt.Println("nonfloatBMI is", nonFloatBmi)
+	// TODO: check and fix syntax for name and bmi variable
+	fmt.Println("BMI is", bmi)
```

Comment:
@nobe4 Tried to get a result by using float variables and multiplying the result by `*10000`.
For some reason it only seems to work with the height in float numbers.

Also tried to convert the height the height converted to a float variable to be able to then divide by 100.
I get`declared and not used: heightAdjustcompiler` even though I'm trying to use it in line 221.
How should I proceed?

Reply:
- using `float64` for all numerical values is a good idea, you can change the map and function accordingly. But you can also just convert the `int` to `float64` like you've done here.
- `Also tried to convert the height the height converted to a float variable to be able to then divide by 100.` => I have no idea what you're trying to say haha
- `declared and not used: heightAdjust` is showing up because you're only ever setting the value of this variable, never using it.